webapp/tests/karma/ts/services/rules-engine.service.spec.ts

```diff
@@ -0,0 +1,619 @@
+    service = TestBed.inject(RulesEngineService);
+
+    service.fetchTargets();
+    service.fetchTaskDocsForAllContacts();
+    service.fetchTaskDocsFor(['a']);
+    await service.isEnabled();
+    
+    expect(fetchTargets.events).to.have.keys(['queued', 'running']);
+    expect(fetchTasksFor.events).to.have.keys(['queued', 'running']);
+    expect(telemetryService.record.callCount).to.equal(6);
+    expect(telemetryService.record.args.map(arg => arg[0])).to.deep.equal([
+      'rules-engine:initialize',
+      'rules-engine:targets:dirty-contacts',
+      'rules-engine:ensureTargetFreshness:cancel',
+      'rules-engine:tasks:dirty-contacts',
+      'rules-engine:ensureTaskFreshness:cancel',
+      'rules-engine:tasks:dirty-contacts',
+    ]);
+
+    fetchTargets.events.queued[0]();
+    fetchTasksFor.events.queued[0]();
+    fetchTasksFor.events.queued[1]();
+
+    expect(telemetryService.record.callCount).to.equal(6);
+    fetchTargets.events.running[0]();
+    expect(telemetryService.record.callCount).to.equal(7);
+    expect(telemetryService.record.args[6]).to.deep.equal(['rules-engine:targets:queued', 0]);
+
+    await Promise.resolve();
+    clock.tick(5000);
+    fetchTargetResultPromise([]);
+    await nextTick();
+
+    expect(telemetryService.record.callCount).to.equal(8);
+    expect(telemetryService.record.args[7]).to.deep.equal(['rules-engine:targets', 5000]);
+    fetchTasksFor.events.running[0]();
+    expect(telemetryService.record.callCount).to.equal(9);
+    expect(telemetryService.record.args[8]).to.deep.equal(['rules-engine:tasks:all-contacts:queued', 5000]);
+    clock.tick(10000);
+    fetchTasksResultPromise[1]([]);
+
+    await nextTick();
+    expect(telemetryService.record.callCount).to.equal(10);
+    expect(telemetryService.record.args[9]).to.deep.equal(['rules-engine:tasks:all-contacts', 10000]);
+    fetchTasksFor.events.running[1]();
+    expect(telemetryService.record.callCount).to.equal(11);
+    expect(telemetryService.record.args[10]).to.deep.equal(['rules-engine:tasks:some-contacts:queued', 15000]);
+    clock.tick(550);
+    fetchTasksResultPromise[3]([]);
+
+    await nextTick();
+    expect(telemetryService.record.callCount).to.equal(12);
+    expect(telemetryService.record.args[11]).to.deep.equal(['rules-engine:tasks:some-contacts', 550]);
+    clock.restore();
+  });
+
+  it('should record correct telemetry data for disabled actions', async () => {
+    authService.has.withArgs('can_view_tasks').resolves(false);
+    const realSetTimeout = setTimeout;
+    const nextTick = () => new Promise(resolve => realSetTimeout(() => resolve()));
+    const clock = sinon.useFakeTimers(1000);
+    fetchTargetsResult = sinon.stub().resolves([]);
+    fetchTasksResult = sinon.stub().resolves([]);
+    service = TestBed.inject(RulesEngineService);
+
+    service.fetchTargets();
+    service.fetchTaskDocsForAllContacts();
+    service.fetchTaskDocsFor(['a']);
+    await service.isEnabled();
```

Comment:
Same as above, the original test called `isEnabled` before the other calls.

Reply:
Fixed